src/Containers/StoreHoursList/CreateHoursModal.tsx

```diff
@@ -13,223 +15,310 @@ import { Select } from '../../Inputs/Select/Select';
+     * @returns {IHoursByDay} - Returns the merged hours
      */
-    const onlyOneTimePerDay = (categoryName: string): boolean => {
-        let check = false;
-        const timedArray = allCategories.find(
-            (
-                categorySchedule: ICategoryWithHoursTypes,
-            ): ICategoryWithHoursTypes | null | boolean =>
-                categorySchedule.category === categoryName,
-        );
-        if (timedArray !== undefined) {
-            Object.entries(checkboxes).forEach((checkbox): void => {
-                if (
-                    checkbox[CHECKBOX_TIME] &&
-                    timedArray.hoursByDay[checkbox[CHECKBOX_DAY]].length >= 1
-                ) {
-                    // cannot have more than one time per day
-                    check = true;
-                }
-            });
+    const mergeOrAddTime = () => {
+        const category = allCategories[addStoreHoursCategory]
+        let merged = false;
+        // unMerged Hours are hours without conflict and always kept
+        const unMergedHoursByDay = {
+            monday: [],
+            tuesday: [],
+            wednesday: [],
+            thursday: [],
+            friday: [],
+            saturday: [],
+            sunday: [],
         }
-        return check;
-    };
 
-    /**
-     * Saves selected time from user
-     * @param {string} categoryName - Name of category that needs hours to be saved
-     * @returns {ICategoryWithHoursType | null} - Updated category schedule with new time
-     */
-    const saveHours = (categoryName: string): void => {
-        const timedArray = allCategories.find(
-            (
-                categorySchedule: ICategoryWithHoursTypes,
-            ): ICategoryWithHoursTypes | null | boolean =>
-                categorySchedule.category === categoryName,
-        );
-        if (timedArray !== undefined) {
-            Object.entries(checkboxes).map(
-                (checkbox): ICategoryWithHoursTypes | null => {
-                    if (
-                        checkbox[CHECKBOX_TIME] &&
-                        timedArray.hoursByDay[checkbox[CHECKBOX_DAY]].length ===
-                            0
-                    ) {
-                        // add time
-                        const convertedTime = convertDateToHours(storeHours);
-                        if (convertedTime !== null) {
-                            timedArray.hoursByDay[checkbox[CHECKBOX_DAY]].push({
-                                from: convertedTime.from,
-                                to: convertedTime.to,
-                            });
-                            setCheckboxes(initialCheckboxState);
-                            return timedArray;
-                        }
-                    }
-                    return null;
-                },
-            );
+        // overWritten Hours are hours that would be lost on merge
+        const overWrittenHoursByDay = {
```

Comment:
Abstract this empty state into an object

Reply:
The reason this is like this is it appeared as though all were referencing the same object in memory.

Will use deepCopy to solve this issue.